iceoryx_posh/include/iceoryx_posh/experimental/popo/base_subscriber.hpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_EXPERIMENTAL_POSH_POPO_BASE_SUBSCRIBER_HPP
+#define IOX_EXPERIMENTAL_POSH_POPO_BASE_SUBSCRIBER_HPP
+
+#include "iceoryx_posh/experimental/popo/sample.hpp"
+#include "iceoryx_posh/internal/popo/ports/subscriber_port_user.hpp"
+#include "iceoryx_posh/popo/condition.hpp"
+#include "iceoryx_utils/cxx/expected.hpp"
+#include "iceoryx_utils/cxx/optional.hpp"
+#include "iceoryx_utils/cxx/unique_ptr.hpp"
+
+namespace iox
+{
+namespace popo
+{
+using uid_t = UniquePortId;
+
+enum class SubscriberError : uint8_t
+{
+    UNKNOWN
```

Comment:
Hehehe ... I would like to rename it into `WHOOPSIE`. Or `MAGICAL_THINGS_JUST_HAPPENED`

Reply:
Ah, I can delete this now, I was using this when I was using the old ports, which didn't have error handling.